### PR TITLE
Divided logic of GetById & PasswordHasher as Service

### LINKED ISSUE #550: Dependency Injection - Password hasher
Get rid of static helpers!
Add this helper to the service collection and inject only where needed.

### LINKED ISSUE #551: Single Responsibility - User Service GetById
We have a method that calls everything from the database and uses it everywhere.
It would be good to divide his responsibility into several new methods.

AuthService GetCurrentUser &
UserService GetById

### PR DESCRIPTION
**Pull request by dyats**
Divided logic of GetById & PasswordHasher as Service

dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)

## Code reviewers

- [ ] @iuriikhrystiuk 

### Second Level Review

- [ ] @sand0 

## Summary of issue

Divide user's fetch logic from database into different methods and use it only where it needs
PasswordHasher is static class and can be used everywhere

## Summary of change

Created methods GetCurrentUserId & GetCurrentAccountId with SecurityContext
PassworhHasher injecting as a service right now

## Testing approach

Fixed unit-tests after code refactoring

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EventsExpress.Core/Services/AuthService.cs [hunk 1]
```diff
@@ -125,22 +135,22 @@ public async Task<AuthenticateResponseModel> EmailConfirmAndAuthenticate(Guid au
             return new AuthenticateResponseModel(jwtToken, refreshToken.Token);
         }
 
-        public async Task ChangePasswordAsync(ClaimsPrincipal userClaims, string oldPassword, string newPassword)
+        public async Task ChangePasswordAsync(string oldPassword, string newPassword)
         {
-            var userDto = GetCurrentUser(userClaims);
+            var userDto = GetCurrentUser();
```
Comment (sand0) [inline EventsExpress.Core/Services/AuthService.cs]: I guess we do not need the whole User object here.
Account with its AuthLocal will be enough.

File: EventsExpress.Core/Services/EventScheduleService.cs [hunk 2]
```diff
@@ -105,7 +103,7 @@ public async Task<Guid> CancelNextEvent(Guid eventId)
             return await Edit(eventScheduleDTO);
         }
 
-        private UserDto CurrentUser() =>
-           _authService.GetCurrentUser(_httpContextAccessor.HttpContext.User);
+        private Guid CurrentUserId() =>
```
Comment (sand0) [inline EventsExpress.Core/Services/EventScheduleService.cs:107]: Could we use the new SecurityContext here? It does the same thing but hasn't a lot of dependencies (as Auth service have)

Comment (dyats) [reply EventsExpress.Core/Services/EventScheduleService.cs:107]: We can but then will need to rewrite a lot of files to change all this stuff(AuthService to SecurityContext).
But still it authService uses SecurityContext to get userId

File: EventsExpress.Core/Services/AuthService.cs [hunk 3]
```diff
@@ -234,16 +241,16 @@ private async Task<Account> ConfirmEmail(CacheDto cacheDto)
             return authLocal.Account;
         }
 
-        public Guid GetCurrUserId(ClaimsPrincipal userClaims)
+        public UserDto GetCurrentUser()
         {
-            Claim guidClaim = userClaims.FindFirst(ClaimTypes.Name);
+            Claim guidClaim = _httpContextAccessor.HttpContext.User.FindFirst(ClaimTypes.Name);
```
Comment (iuriikhrystiuk) [inline EventsExpress.Core/Services/AuthService.cs]: Why did not we use ISecurityContext here and remove dependency on HttpContextAccessor?

File: EventsExpress/appsettings.json [hunk 4]
```diff
@@ -24,7 +24,7 @@
         "JwtEmailEncryption": "YOUR JWT EMAIL ENCRYPTION"
     },
   "ConnectionStrings": {
-    "DefaultConnection": "Server=(localdb)\\mssqllocaldb;Database=EventsExpress;Trusted_Connection=True;MultipleActiveResultSets=true",
```
Comment (iuriikhrystiuk) [inline EventsExpress/appsettings.json]: Undo this file please.

**Review by iuriikhrystiuk — APPROVED** [review]

**Pull request merged**